src/features/network/network-provider.tsx

```diff
@@ -7,39 +7,42 @@ import {
   Account,
   Base,
   Events,
-} from "@liftedinit/many-js"
-import { useNetworkStore } from "./store"
-import { useAccountsStore } from "features/accounts"
+  Tokens,
+} from "@liftedinit/many-js";
+import { useNetworkStore } from "./store";
+import { useAccountsStore } from "features/accounts";
 
 const NetworkContext = React.createContext<[Network?, Network?]>([
   undefined,
   undefined,
-])
+]);
 
 export function NetworkProvider({ children }: React.PropsWithChildren<{}>) {
-  const activeNetwork = useNetworkStore(state => state.byId.get(state.activeId))
-  const activeAccount = useAccountsStore(state =>
-    state.byId.get(state.activeId),
-  )!
+  const activeNetwork = useNetworkStore((state) =>
+    state.byId.get(state.activeId)
+  );
+  const activeAccount = useAccountsStore((state) =>
+    state.byId.get(state.activeId)
+  )!;
 
   const network = React.useMemo(() => {
-    const anonIdentity = new AnonymousIdentity()
-    const identity = activeAccount?.identity ?? anonIdentity
-    const url = activeNetwork?.url || ""
-    const queryNetwork = new Network(url, anonIdentity)
-    queryNetwork.apply([Ledger, IdStore, Account, Events, Base])
-    const cmdNetwork = new Network(url, identity)
-    cmdNetwork.apply([Ledger, IdStore, Account])
-    return [queryNetwork, cmdNetwork] as [Network, Network]
-  }, [activeNetwork, activeAccount])
+    const anonIdentity = new AnonymousIdentity();
+    const identity = activeAccount?.identity ?? anonIdentity;
+    const url = activeNetwork?.url || "";
+    const queryNetwork = new Network(url, anonIdentity);
+    queryNetwork.apply([Ledger, IdStore, Account, Events, Base, Tokens]);
+    const cmdNetwork = new Network(url, identity);
+    cmdNetwork.apply([Ledger, IdStore, Account, Tokens]);
+    return [queryNetwork, cmdNetwork] as [Network, Network];
```

Comment:
This is existing code and also (IMHO) a little weird. We create two different network connections, an anonymous one for queries and an authenticated one for mutations. Are queries ever going to be permissioned?

Reply:
They could be.

Reply:
I don't want to reinvent everything that Tommy did just to "make it my own" or anything, but I do think it needs a second look from the perspective of the app developer. The JS library should make app development really simple and straightforward and I'm not 100% sure it does in its current form.